### PR TITLE
fix: Clicking on room name redirect to connected device fragment

### LINKED ISSUE #2286: Clicking on room name redirects to wrong page
**Describe the bug**
Currently, if you click the room name it redirects to connectedDeviceFragment instead of DeviceConnectFragment. The error occurred during the resolving of conflicts. An intent parameter is missing.

**Expected Behaviour**
Clicking on room name should redirect to DeviceConnectFragment.

**Steps to reproduce it**
Click on a room name in the page that opens connect to device page.

**Smartphone Info:**

|               | |
|---------------|-|
|Device         | RedMi Note 5 pro|
|Android Version | Pi 9.0   |

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by atm1504**
Fixes #2286

Changes:

Screenshots for the change: 
ezgif com-video-to-gif

Apk: 
[app-fdroid-debug.apk.zip](https://github.com/fossasia/susi_android/files/3341879/app-fdroid-debug.apk.zip)

### CODE REVIEW

**Comment by atm1504** [comment]
@iamareebjamal please review it. It is an important bug that needs to be fixed.

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/adapters/recycleradapters/RoomsAdapter.kt [hunk 1]
```diff
@@ -34,6 +34,7 @@ class RoomsAdapter(private val availableRoomsList: ArrayList<DeviceConnectFragme
                 val roomNameClicked = availableRoomsList[adapterPosition].room
                 var intent = Intent(context, DeviceActivity::class.java)
                 intent.putExtra("roomName", roomNameClicked)
+                intent.putExtra("connectTo", "DeviceConnectFragment")
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/adapters/recycleradapters/RoomsAdapter.kt]: Why is it used?

Comment (atm1504) [reply app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/adapters/recycleradapters/RoomsAdapter.kt]: The DeviceActivity is the base activity for 2 major features. The intent parameter is passed to tell the activity which fragment it should load.

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/adapters/recycleradapters/RoomsAdapter.kt]: Make it a constant

Comment (atm1504) [reply app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/adapters/recycleradapters/RoomsAdapter.kt]: Updated

File: app/src/main/java/org/fossasia/susi/ai/helper/Constant.kt [hunk 2]
```diff
@@ -15,6 +15,9 @@ object Constant {
     const val MIC_INPUT = "mic_input"
 
     const val ACCESS_TOKEN = "access_token"
+    const val CONNECT_TO = "connectTo"
+    const val DEVICE_CONNECT_FRAGMENT = "device_connect_fragment"
+    const val CONNECTED_DEVICE_FRAGMENT = "connected_device_fragment"
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/helper/Constant.kt]: Move to the activity

Comment (atm1504) [reply app/src/main/java/org/fossasia/susi/ai/helper/Constant.kt]: You mean to rename the variables?

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/helper/Constant.kt]: No, move them

File: app/src/main/java/org/fossasia/susi/ai/skills/settings/ChatSettingsFragment.kt [hunk 3]
```diff
@@ -48,6 +48,8 @@ class ChatSettingsFragment : PreferenceFragmentCompat(), ISettingsView {
     private val TAG_ABOUT_FRAGMENT = "AboutUsFragment"
     private val TAG_HELP_FRAGMENT = "HelpFragment"
     private val TAG_PRIVACY_FRAGMENT = "PrivacyFragment"
+    private val TAG_CONNECTED_DEVICE_FRAGMENT = "ConnectedDeviceFragment"
+    private val TAG_DEVICE_CONNECT_FRAGMENT = "DeviceConnectFragment"
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/skills/settings/ChatSettingsFragment.kt]: When you duplicate the variable, the entire point of making it a constant is lost

**Comment by atm1504** [comment]
@iamareebjamal please review it.

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/adapters/recycleradapters/RoomsAdapter.kt [hunk 4]
```diff
@@ -34,6 +34,7 @@ class RoomsAdapter(private val availableRoomsList: ArrayList<DeviceConnectFragme
                 val roomNameClicked = availableRoomsList[adapterPosition].room
                 var intent = Intent(context, DeviceActivity::class.java)
                 intent.putExtra("roomName", roomNameClicked)
+                intent.putExtra(DeviceActivity().CONNECT_TO, DeviceActivity().TAG_DEVICE_CONNECT_FRAGMENT)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/adapters/recycleradapters/RoomsAdapter.kt]: Wow

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/adapters/recycleradapters/RoomsAdapter.kt]: No need to instantiate activity. Heard of static final variables?

Comment (atm1504) [reply app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/adapters/recycleradapters/RoomsAdapter.kt]: Yes, we can do so via using the companion object in kotlin.

**Pull request merged**